### PR TITLE
Constraints panel component

### PR DESCRIPTION
**Pull request by lattemilk**
Note: Subject to change due to new mocks

# Changes:
- Added constraints panel component
- Added "add constraints" step to "create a new scenario"
- Used angular form control and form validation
- Selects all treatment types and land ownership types by default

# UI Updates: 
Screenshot 2023-01-19 at 2 54 57 PM

# Todo:
- Populate existing constraints from existing scenario
- Update the progress panel to follow steps
- Save constraints and priorities on "next" button click

### CODE REVIEW

File: src/interface/src/app/plan/create-scenarios/create-scenarios.component.html [hunk 1]
```diff
@@ -2,6 +2,7 @@
   <div class="create-scenarios-panel-content" [@expandCollapsePanelContent]="panelExpanded ? 'opaque' : 'transparent'">
     <app-create-scenarios-intro *ngIf="planningStep === 1"></app-create-scenarios-intro>
     <app-set-priorities *ngIf="planningStep === 2" (changeConditionEvent)="changeConditionEvent.emit($event)"></app-set-priorities>
+    <app-constraints-panel *ngIf="planningStep === PlanStep.SetConstraints" (changeConstraintsEvent)="changeConstraintsEvent.emit($event)"></app-constraints-panel>
```
Comment (thequeenofspades) [inline src/interface/src/app/plan/create-scenarios/create-scenarios.component.html:5]: do you mind changing lines 3 and 4 to use the PlanStep type as well?

File: src/interface/src/app/plan/create-scenarios/constraints-panel/constraints-panel.component.html [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+          <mat-icon class="material-symbols-outlined">info</mat-icon>
+        </mat-chip-avatar>
+        Required
+      </mat-chip>
+    </mat-chip-list>
+  </div>
+
+  <!-- Description -->
+  <p class="constraints-description">
+    Constraints will be used to inform project areas and scenario options
```
Comment (thequeenofspades) [inline src/interface/src/app/plan/create-scenarios/constraints-panel/constraints-panel.component.html:17]: nit: add period

File: src/interface/src/app/plan/create-scenarios/constraints-panel/constraints-panel.component.html [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+  <!-- Form -->
+  <div class="constraints-content">
+    <form class="constraints-form" [formGroup]="constraintsForm">
+      <!-- Budget -->
+      <div formGroupName="budgetForm" required>
+        <label class="form-label required">Budget</label>
+        <mat-radio-group formControlName="optimizeBudget" required>
+          <!-- Custom Budget -->
+            <mat-radio-button [value]="false">
+              <!-- Min -->
```
Comment (thequeenofspades) [inline src/interface/src/app/plan/create-scenarios/constraints-panel/constraints-panel.component.html:29]: The min and max fields don't appear in the screenshot you took. Were they added after?

**Review by thequeenofspades — APPROVED** [review]
very nice!!

File: src/interface/src/app/plan/create-scenarios/constraints-panel/constraints-panel.component.html (hunk 3 above)
Comment (lattemilk) [reply src/interface/src/app/plan/create-scenarios/constraints-panel/constraints-panel.component.html:29]: Oops I took a screenshot with it scrolled down. Updated!

File: src/interface/src/app/plan/create-scenarios/create-scenarios.component.html (hunk 1 above)
Comment (lattemilk) [reply src/interface/src/app/plan/create-scenarios/create-scenarios.component.html:5]: done!

File: src/interface/src/app/plan/create-scenarios/constraints-panel/constraints-panel.component.html (hunk 2 above)
Comment (lattemilk) [reply src/interface/src/app/plan/create-scenarios/constraints-panel/constraints-panel.component.html]: done.

**Pull request merged**